src/simple.jl

```diff
@@ -121,12 +122,16 @@ function cvode(f::Function, y0::Vector{Float64}, t::Vector{Float64}, userdata::A
         tout = [0.0]
         for k in 2:length(t)
             flag = @checkflag CVode(mem, t[k], ynv, tout, CV_NORMAL)
+            if !callback(ynv)
```

Comment:
Generalize in what way? With clever uses of closures you can already accomplish quite a bit with this implementation.